### DISCUSSION TITLE
Portal :: Angular Migration to version 12

### DISCUSSION THREAD

**Discussion by rajeshkumaravel** [Sunbird-Ed/Community#333] [Contribution]
- Sunbird ED is planning to update the angular version from 11 to 12.
- Below are the Repos and the team owning the same.

| **Repo**                                      | **Owned By** |
|-----------------------------------------------|--------------|
| SunbirdED-portal                              | SunbirdED    |
| @project-sunbird/chatbot-client-v8            | SunbirdED    |
| @project-sunbird/common-consumption-v9        | SunbirdED    |
| @project-sunbird/common-form-elements-v9      | SunbirdED    |
| @project-sunbird/discussions-ui-v8            |
| @project-sunbird/sunbird-epub-player-v9       | Player Team  |
| @project-sunbird/sb-dashlet-v9                | SunbirdED    |
| @project-sunbird/sunbird-collection-editor-v9 | Knowledge    |
| @project-sunbird/sunbird-pdf-player-v9        | Player Team  |
| @project-sunbird/sunbird-question-editor      | Knowledge    |
| @project-sunbird/sunbird-quml-player-v9       | Player Team  |
| @project-sunbird/sunbird-video-player-v9      | Player Team  |
| @project-sunbird/web-extensions               | SunbirdED    |
| @samagra-x/uci-console                        | Samagra      |
| @shikshalokam/sl-questionnaire                | ShikshaLokam |
| @shikshalokam/sl-reports-library              | ShikshaLokam |
| @swimlane/ngx-datatable                       |
| common-form-elements-v9                       | SunbirdED    |
| common-form-elements-web-v9                   | SunbirdED    |
| sb-notification                               | SunbirdED    |
| sb-svg2pdf                                    | Mobile Team  |

---
- Status: https://docs.google.com/spreadsheets/d/1CPlFSXWKAHU-InKUVSVPvr775mGrOuFTau4Qd7z8CLQ/edit#gid=0
---

@vinukumar-vs @rajeevsathish @Krishnaj20  @sharathkashyap

**Comment by Krishnaj20** [comment]
tagging others: @pallakartheekreddy @rhwarrier @reshmi-nair @anandp504 @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @Prerna789 @pkranga @SanthoshVasabhaktula @vrayulu @ashokreddy1208 @gandham-santhosh @kameshbhr @mmanimegalai @coolbung @parthlawate 

The work for this migration has started in 5.1.1 it is expected to be made live as part of 5.3.0 release.

**Reply by HarishGangula** [reply]
@rajeshkumaravel  sb-svg2pdf is not belong to player team, please update it accordingly all the players are frontend framework agnostic since players are exported as web components refer [here](https://knowlg.sunbird.org/learn/product-and-developer-guide/player) for details
cc: @Krishnaj20  @vinukumar-vs @pallakartheekreddy

**Reply by rajeshkumaravel** [reply]
@HarishGangula List is updated.
As discussed with @swayangjit , assigning sb-svg2pdf to mobile team.
cc: @Krishnaj20 @vinukumar-vs @rajeevsathish

**Reply by HarishGangula** [reply]
Sorry i missed to mention @project-sunbird/sunbird-quml-player-v9 belongs to [inquiry](https://inquiry.sunbird.org/learn/readme) BB please update that as well @rajeshkumaravel

**Comment by parthlawate** [comment]
@rajeshkumaravel Any study done on checking effort for moving from 11->12 vs directly moving to 14 (ideally 15) ?  At a minimum should we not be moving to a version that is officially supported by the community ? 

reference : https://angular.io/guide/releases#actively-supported-versions

If we dont, we will end up having to support issues which might be there in v12 which we will need to end up supporting as a community.. 

@Krishnaj20 this is also relevant to the other version /support policy we were discussing.

**Reply by SanthoshVasabhaktula** [reply]
@rajeshkumaravel - What is the ionic version used by mobile app?

@parthlawate - Angular version update may not be that simple as it seems as there are quite a few components which are common across mobile app and portal. To support Angular 12+ ionic version should also be upgraded.

**Reply by parthlawate** [reply]
@SanthoshVasabhaktula Yes I understand.. However as a community strategy, to reduce support load on us as a community, if we can try to be on LTS releases of upstream libraries it will help us improver stability and also reduce support load. 

Reference IONIC Support policy & supported versions https://ionicframework.com/docs/reference/support

**Reply by SanthoshVasabhaktula** [reply]
Yes - that was always the intent and we need community members extensive support to be on LTS releases of upstream libraries.

**Reply by parthlawate** [reply]
Ok thats awesome.. If we are not able to commit to the latest LTS's immediately.. could we define it as roadmap items and publish them ?

**Reply by rajeevsathish** [reply]
@SanthoshVasabhaktula as @rajeshkumaravel is in portal team we do not have complete knowledge about the ionic version. As checked with the mobile team the current ionic version is 5.4.16

**Reply by Krishnaj20** [reply]
continuing from https://github.com/orgs/Sunbird-knowlg/discussions/87
@arun1205 - which component are you referring to?
@mmanimegalai @vinukumar-vs
